api/controller.go

```diff
@@ -2372,6 +2396,68 @@ func (c *Controller) DetachPolicyFromGroupHandler() authop.DetachPolicyFromGroup
 	})
 }
 
+func (c *Controller) RefsRefsRestoreHandler() refs.RefsRestoreHandler {
+	return refs.RefsRestoreHandlerFunc(func(params refs.RefsRestoreParams, user *models.User) middleware.Responder {
+		deps, err := c.setupRequest(user, params.HTTPRequest, []permissions.Permission{
+			{
+				Action:   permissions.ListTagsAction,
+				Resource: permissions.RepoArn(params.Repository),
+			},
+			{
+				Action:   permissions.ListBranchesAction,
+				Resource: permissions.RepoArn(params.Repository),
+			},
+			{
+				Action:   permissions.ListCommitsAction,
+				Resource: permissions.RepoArn(params.Repository),
+			},
```

Comment:
I think permission actions are wrong here: this is a modifying operation, I think we should test some action that says "Write".

Reply:
fixed